File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/database/ContactDatabase.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Contacts.class}, version = 1)
public abstract class ContactDatabase extends RoomDatabase {
    public abstract ContactDAO getContactDAO();

    //Singleton pattern(Ensuring you only have one instance of the db)
    public static ContactDatabase dbInstance;
    public static synchronized ContactDatabase getInstance(Context context){
        if (dbInstance == null){
            dbInstance = Room.databaseBuilder(
                    context.getApplicationContext(),
                    ContactDatabase.class, "contacts_db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return dbInstance;
    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/database/ContactDatabase.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Contacts::class], version = 2)
abstract class ContactDatabase : RoomDatabase() {

    abstract fun getContactDAO(): ContactDAO

    companion object {
        @Volatile
        private var dbInstance: ContactDatabase? = null

        fun getInstance(context: Context): ContactDatabase {
            return dbInstance ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    ContactDatabase::class.java,
                    "contacts_db"
                ).fallbackToDestructiveMigration()
                    .build()
                dbInstance = instance
                instance
            }
        }
    }
}